cmd/snaplock/runinhibit/inhibit_test.go

```diff
@@ -273,20 +279,70 @@ func (s *runInhibitSuite) TestWaitWhileInhibitedWalkthrough(c *C) {
 	flock.Close()
 }
 
+func (s *runInhibitSuite) TestWaitWhileInhibitedContextCancellation(c *C) {
+	c.Assert(runinhibit.LockWithHint("pkg", runinhibit.HintInhibitedForRefresh, s.inhibitInfo), IsNil)
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	inhibitedCalled := 0
+	tickerWait := func() <-chan time.Time {
+		// lock should be released during wait interval
+		checkFileNotLocked(c, runinhibit.HintFile("pkg"))
+
+		if inhibitedCalled == 2 {
+			// let's cancel the context
+			cancel()
+		}
+
+		// give precedence to ctx.Done() channel
+		return time.After(50 * time.Millisecond)
```

Comment:
this can be longer no for safety?

Reply:
it's always safer to increase the timeout, updated, thanks!